### DISCUSSION TITLE
Override Notehead on Beam?

### DISCUSSION THREAD

**Discussion by nicholasjconn** [0xfe/vexflow#1416] [Q&A] [answered]
I'm trying to render the Raindrop Prelude and am running into a problem. I can't figure out how to render a half note on the beam (see image below).

<img width="280" alt="image" src="https://user-images.githubusercontent.com/350756/175751442-dc6190cb-67d2-475c-8e94-0d6e5efddc0a.png">

Any suggestions? Would it be possible to create it with all eighth notes and then override the glyphs somehow to render as a half note?

**Comment by rvilarl** [comment]
Perhaps you can use glyphnote

**Comment by AaronDavidNewman** [comment]
>> perhaps you can use glyphnote

There's no way to set the head directly?  We could add the 'normal' note heads to `Tables.customNoteHeads`, then it would work like notehead_tests.

btw @nicholasjconn  what is the interpretation of this custom note head?  I've not seen it before.  I've seen diamond note heads for tremolo.

**Comment by AaronDavidNewman** [comment, marked as answer]
I went down the rabbit hole and was able to get it to work.  In EasyScore, creating a new 'valid code' and then adding it to Tables.durationCodes.  This table is kind of a mess - you have to add it for each duration.  IMO there should be a more general way of handling note heads.

![image](https://user-images.githubusercontent.com/5438280/175795001-b8e7c280-8fba-4c33-8103-9fbd56429a19.png)

beam_tests.ts

``` typescript
const voice = score.voice(
    [
    ...
      beam(notes('(cb4 e#4 a4)/8/z, (d4 f4 a4)/8, (ebb4 g##4 b4)/8, (f4 a4 c5)/8', { stem: 'up' })),
```

tables.ts:

``` typescript
const durationCodes: Record<string, any> = {
...
  '8': { // just for eighth notes
     type: {
...
       z: {code_head: 'noteheadHalf'}
```

**Reply by nicholasjconn** [reply]
Wow! This is a great workaround! Thank you so much @AaronDavidNewman!

The interpretation (at least, on a piano), is that the half notes are held down while the other eighth notes are played. The image below is the workaround I used.

<img width="220" alt="image" src="https://user-images.githubusercontent.com/350756/175795859-4bc45e81-e432-4f17-93f9-5f964f2b940b.png">

I've also seen it noted like this.

<img width="156" alt="image" src="https://user-images.githubusercontent.com/350756/175795957-2ff43fdd-b1fc-4476-ae01-c83a1619082f.png">

I really appreciate your help on this.

**Reply by rvilarl** [reply]
@AaronDavidNewman this last two examples are more related to sharing stem on formatter, are not they?

**Reply by AaronDavidNewman** [reply]
I'm glad this helped.  Feel free to submit a PR if you think this will be generally useful.  Maybe we should use a different letter instead of 'z' - maybe 'o' for open?   And it can be added to all the closed-note durations (>= '8') in durationCodes.

It should work like your second example if you use a second voice.  @rvilarl  is it possible to get the head to go in the other direction?  I could manually if I create the StaveNote directly but not with easy score.

**Reply by AaronDavidNewman** [reply]
@rvilarl  you're right that would be sharing stem.  I was just thinking that voices should do that automatically if they share a note with different duration that starts in the same place.

**Reply by rvilarl** [reply]
I think that there is a condition that checks that the duration is the same to allow sharing the stem. I will have a look.

**Reply by nicholasjconn** [reply]
I'll plan to create a PR for it sometime in the next few weeks. I'm working on the first release of my app and am moving forward with my workaround for now. Thank you both again.
